### ISSUE TITLE
GHC 8.8 support

### ISSUE THREAD

**Issue by kozross** [expipiplus1/vector-sized#78]
Currently, this won't even _configure_, much less build:

```
[koz@Sebastian vector-sized]$ cabal new-configure -w /home/koz/ghc-88/bin/ghc
'cabal.project.local' file already exists. Now overwriting it.
Warning: Parsing the index cache failed (Unknown encoding for constructor).
Trying to regenerate the index cache...
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: vector-sized-1.3.0.0 (user goal)
[__1] trying: indexed-list-literals-0.2.1.2 (dependency of vector-sized)
[__2] next goal: base (dependency of vector-sized)
[__2] rejecting: base-4.13.0.0/installed-4.1... (conflict:
indexed-list-literals => base>=4.9 && <4.13)
[__2] rejecting: base-4.12.0.0, base-4.11.1.0, base-4.11.0.0, base-4.10.1.0,
base-4.10.0.0, base-4.9.1.0, base-4.9.0.0, base-4.8.2.0, base-4.8.1.0,
base-4.8.0.0, base-4.7.0.2, base-4.7.0.1, base-4.7.0.0, base-4.6.0.1,
base-4.6.0.0, base-4.5.1.0, base-4.5.0.0, base-4.4.1.0, base-4.4.0.0,
base-4.3.1.0, base-4.3.0.0, base-4.2.0.2, base-4.2.0.1, base-4.2.0.0,
base-4.1.0.0, base-4.0.0.0, base-3.0.3.2, base-3.0.3.1 (constraint from
non-upgradeable package requires installed instance)
[__2] fail (backjumping, conflict set: base, indexed-list-literals,
vector-sized)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, vector-sized,
indexed-list-literals
```

Edit: @mstksg suggested using ``--allow-newer``. Using that, I was able to get a successful build, which seems to indicate overly pessimistic bounds on ``base`` in ``indexed-list-literals``.

**Comment by kozross** [comment]
I've made a PR, but the owner seems unresponsive. What do we actually use ``indexed-list-literals`` for? I'd be willing to rewrite this if it gets this issue solved quicker.

**Comment by mstksg** [comment]
We use `indexed-list-literals` to implement `fromTuple`:

```haskell
fromTuple :: (a,a) -> Vector 2 a
fromTuple :: (a,a,a) -> Vector 3 a
fromTuple :: (a,a,a,a) -> Vector 4 a
-- etc.
```

The typeclass-based mechanism comes from `indexed-list-literals`, so it would probably be more ideal to get it from there.  Maybe we could give it a little more time and see if the PR is accepted?

**Comment by kozross** [comment]
@mstksg Ah, I see. Sure, let's wait a bit - my project's stalled on GHC 8.8 support until Hedgehog gets it anyway.

**Comment by kozross** [comment]
OK, we're unblocked now - we need to up the bound on ``indexed-list-literals`` to ``0.2.1.2``, and then we can build on GHC 8.8 without troubles. Also, thanks @gwils!

**Comment by expipiplus1** [comment]
Released as v1.4.0.0
